app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -164,4 +170,45 @@ class MessagesController @Inject() (
     case _                     => EitherT.leftT(PresentationError.badRequestError(s"${messageType.code} is not valid for requests"))
   }
 
+  private def updateMessageStatus(
+    sdesResponse: SdesNotificationItem,
+    messageStatus: MessageStatus
+  )(implicit
+    hc: HeaderCarrier
+  ) = {
+    val (movementId, messageId) = extractMovementMessageId(sdesResponse)
+    (for {
+      persistenceResponse <- persistenceConnector
+        .patchMessageStatus(movementId, messageId, MessageUpdate(messageStatus))
+        .asPresentation
+
+    } yield persistenceResponse).fold[Result](
```

Comment:
Please yield and fold it to inside original method call as we need to implement ppns method call as well so better to do this with ``` handleSdesResponse ```